streams/src/main/java/org/apache/kafka/streams/kstream/internals/ProducedInternal.java

```diff
@@ -36,4 +36,8 @@ public ProducedInternal(final Produced<K, V> produced) {
     public StreamPartitioner<? super K, ? super V> streamPartitioner() {
         return partitioner;
     }
+
+    public String processorName() {
```

Comment:
This is a meta comment: I'd suggest let other XXXInternal to extends XXX and NamedInternal so that this function is consolidated on the base class as `public String name()` which is allowed to return null.

Reply:
I think what you mean is that Produced/Consumed classes should extend Named ? Because XXXInternal classes already extend XXX (no multiple inheritance)

Reply:
Ah yes, you are right. Hmm then that means we need to move this function to `Named` so that:

Named (contains `processorName()`) -> Consumed / etc -> ConsumedInternal / etc (shares `processorName()`

But this is a bit awkward since all other `Consumed / etc` public classes does not have getters only setters.

So how about making this function package-private while moving to Named, so that it will not be exposed as public APIs but still be able to accessed internally on XXXInternal callers?